### PR TITLE
Refactor initial app loading

### PR DESCRIPTION
**Pull request by fhenrich33**

### CODE REVIEW

**Comment by fhenrich33** [comment]
[2.0 Reduce initial loading time by removing auth waterfall in FE](https://trello.com/c/eq3ZB7Ao/1352-20-reduce-initial-loading-time-by-removing-auth-waterfall-in-fe)

File: front/src/providers/ApolloAuth0Provider.tsx [hunk 1]
```diff
@@ -37,12 +37,19 @@ function ApolloAuth0Provider({ children }: { children: ReactNode }) {
 
   const errorLink = onError(({ graphQLErrors, networkError }) => {
     if (graphQLErrors) {
-      graphQLErrors.map(({ message, locations, path }) =>
+      graphQLErrors.map(({ message, locations, path, extensions }) => {
+        // Supress 'base=0' errors. We are trying to preload queries and cache results.
```
Comment (pylipp) [inline front/src/providers/ApolloAuth0Provider.tsx:41]: two things come to my mind here:
1. we must be careful to not change the format of the error message returned by the BE. In some way, the error message becomes part of the interface that the FE relies on
2. Requesting `base=0` will cause [Sentry reports](https://boxwise.sentry.io/issues/5670060810/?alert_rule_id=12109046&alert_type=issue&notification_uuid=c2ef13f7-4ad3-4c94-8b0d-eb339ae4f9bc&project=6584296&referrer=slack) 

Not sure how far we want to go here but we at least the Sentry errors don't have to necessarily be reported

Comment (fhenrich33) [reply front/src/providers/ApolloAuth0Provider.tsx:41]: @pylipp Maybe there's a more nuanced and elegant way of going about this, but I'm only skipping this particular message exactly so as not to flare users with unneeded error messages.

About Sentry:
Making a note to suppress this error here: https://trello.com/c/JWPwlcRL/1462-20-fe-update-sentry-config

Comment (fhenrich33) [reply front/src/providers/ApolloAuth0Provider.tsx]: I think I got a better way. Check it out @pylipp !

File: front/src/components/QrReader/components/QrReaderMultiBoxContainer.tsx [hunk 2]
```diff
@@ -26,10 +25,11 @@ import {
   FailedBoxesFromMoveBoxesAlert,
   NotInStockAlertText,
 } from "./AlertTexts";
+import { useLoadAndSetGlobalPreferences } from "hooks/useLoadAndSetGlobalPreferences";
 
 function QrReaderMultiBoxContainer() {
-  const { globalPreferences } = useContext(GlobalPreferencesContext);
-  const currentBaseId = globalPreferences.selectedBase?.id;
+  const { baseId } = useLoadAndSetGlobalPreferences();
```
Comment (HaGuesto) [inline front/src/components/QrReader/components/QrReaderMultiBoxContainer.tsx:31]: put baseId retrieval into a separate helper function or hook to make the code more readible.

File: front/src/hooks/useLoadAndSetGlobalPreferences.ts [hunk 3]
```diff
@@ -15,6 +16,14 @@ export const useLoadAndSetGlobalPreferences = () => {
 
   // retrieve base id from the url
   const baseIdInput = location.pathname.match(/\/bases\/(\d+)(\/)?/);
+  /**
+   * Infers intended base ID from URL.
+   * 
+   * Defaults to `"0"` if there's no base ID to still run the queries and cache results.
```
Comment (HaGuesto) [inline front/src/hooks/useLoadAndSetGlobalPreferences.ts:22]: clean up comment

File: front/src/hooks/useLoadAndSetGlobalPreferences.ts [hunk 4]
```diff
@@ -15,6 +16,14 @@ export const useLoadAndSetGlobalPreferences = () => {
...
+   * 
+   * We will supress `'base=0'` errors in the `ApolloAuth0Provider` error handler as they originate from this parameter.
+   */
+  const baseId = baseIdInput?.length && baseIdInput[1] || "0";
```
Comment (HaGuesto) [inline front/src/hooks/useLoadAndSetGlobalPreferences.ts:26]: move to seperate function/hook

File: front/src/hooks/useLoadAndSetGlobalPreferences.ts [hunk 5]
```diff
@@ -23,16 +32,27 @@ export const useLoadAndSetGlobalPreferences = () => {
   const [runLocationAndShipmentQuery] = useLazyQuery<MultiBoxActionOptionsForLocationsTagsAndShipmentsQuery>(
     MULTI_BOX_ACTION_OPTIONS_FOR_LOCATIONS_TAGS_AND_SHIPMENTS_QUERY,
     {
-      variables: { baseId: baseIdInput && baseIdInput[1] || "0" },
+      variables: { baseId },
+    },
+  );
+
+  // fetch options for actions on boxes
+  const [runActionOptionsForBoxesView] = useLazyQuery<ActionOptionsForBoxesViewQuery>(
```
Comment (HaGuesto) [inline front/src/hooks/useLoadAndSetGlobalPreferences.ts:40]: remove these queries

File: front/src/views/Boxes/BoxesView.tsx [hunk 6]
```diff
@@ -96,8 +96,7 @@ export const ACTION_OPTIONS_FOR_BOXESVIEW_QUERY = gql`
 `;
 
 function Boxes() {
-  const { globalPreferences } = useContext(GlobalPreferencesContext);
-  const baseId = globalPreferences.selectedBase?.id!;
+  const { baseId } = useLoadAndSetGlobalPreferences();
```
Comment (HaGuesto) [inline front/src/views/Boxes/BoxesView.tsx:99]: put baseId retrieval into a separate helper function or hook to make the code more readible.

**Review by HaGuesto — APPROVED** [review]
I will approve this PR because I trust Felipe to fix the comments of the review himself.

write a follow up trello ticket to replace instances of 

```
  const { globalPreferences } = useContext(GlobalPreferencesContext);
  const currentBaseId = globalPreferences.selectedBase?.id;
  ```
  with a baseid retrieval from the url

**Comment by HaGuesto** [comment]
please write up a ticket for:
- menu randomly opening during initial loading

File: front/src/hooks/useLoadAndSetGlobalPreferences.ts (hunk 5 above)
Comment (fhenrich33) [reply front/src/hooks/useLoadAndSetGlobalPreferences.ts:40]: I think you meant the ones in the routes themselves, we want these to load first in the component tree.

File: front/src/providers/ApolloAuth0Provider.tsx (hunk 1 above)
Comment (pylipp) [reply front/src/providers/ApolloAuth0Provider.tsx]: Yeah makes total sense to avoid running the queries with baseId 0 🚀 thx!

**Pull request merged**